chain/chaintest/balance_handler.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/codec/codectest"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/state/tstate"
+)
+
+// TestBalanceHandler tests b by requiring that it upholds the invariants
+// described in the BalanceHandler interface.
+func TestBalanceHandler(ctx context.Context, t *testing.T, b chain.BalanceHandler) {
+	// Test addresses
+	addrOne := codectest.NewRandomAddress()
+	addrTwo := codectest.NewRandomAddress()
```

Comment:
Another idea is to refactor this so that you can have an array of tests, that take in an array of addresses to run on so you don't have to write separate tests for the multi-address case vs single address case.